storage/s3/src/server/wrappers/S3ClientWrapper.ts

```diff
@@ -99,7 +99,35 @@ export class S3ClientWrapper {
     await upload.done();
   }
 
-  public async list(
+  public async listDirectories(): Promise<BaseDirectory[]> {
+    let truncated: boolean | undefined = true;
+    let continuationToken: string | undefined;
+    let directories: BaseDirectory[] = [];
+    while (truncated) {
+      /* eslint-disable @typescript-eslint/naming-convention */
+      const response = await this._client.send(
+        new ListObjectsV2Command({
+          Bucket: this._bucket,
+          ContinuationToken: continuationToken,
+          Delimiter: "/",
+        })
+      );
+      directories = directories.concat(
```

Comment:
This is a bit hard to read - it is hard to tell what does the `CommonPrefixes` property hold,